chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -203,6 +209,30 @@ def usage
           self.owner
         end
 
+        def multipath?
+          self.class.multipath?(@device)
+        end
+
+        def multipath_path?
+          # we need to check if the holders of a device (if it has holders)
+          # are a multipath, for example:
+          #
+          # 0QEMU_QEMU_HARDDISK_00002 dm-1 QEMU,QEMU HARDDISK
+          # size=10G features='0' hwhandler='0' wp=rw
+          # `-+- policy='service-time 0' prio=1 status=active
+          # `- 0:0:0:2 sdc 8:32   active ready running
+          # `-+- policy='service-time 0' prio=1 status=enabled
+          # `- 0:0:0:3 sdb 8:16   active ready running
+          # sdb and sdc are paths of dm-1:
+          # root@d52-54-77-77-01-01:~ # ls /sys/block/sdb/holders/
+          # dm-1
```

Comment:
This is super helpful - thanks a lot for adding this comment!  Although it looks like some lines got accidentally wrapped.